dev_guide/app_tutorials/ruby_on_rails.adoc

```diff
@@ -0,0 +1,357 @@
+* Creating a route for your application.
+
+=== Creating database service
+
+Our Rails application will expect a running database link:../../architecture/core_concepts/pods_and_services.html#services[service] running. For this service we will use link:http://www.postgresql.org/[PostgeSQL] database link:../../using_images/db_images/postgresql.html[image].
+
+To create the database link:../../architecture/core_concepts/pods_and_services.html#services[service] you will use the link:../new_app.html[oc new-app] command. To this command you will need to pass some necessary link:../../using_images/db_images/postgresql.html#environment-variables[environment variables] which will be used inside the database container. These link:../../db_images/postgresql.html#environment-variables[environment variables] are required to set the username, password, and name of the database. You can change the values of these link:../../db_images/postgresql.html#environment-variables[environment variables] to anything you would like. The variables we are going to be setting are as follows:
+
+* POSTGRESQL_DATABASE
+* POSTGRESQL_USER
+* POSTGRESQL_PASSWORD
+
+By setting these variables when creating the PostgreSQL database, the image will ensure that:
+
+* A database exists with the specified name
+* A user exists with the specified name
+* The user can access the specified database with the specified password
+
+To do that type following command with your desired values:
+
+----
+$ oc new-app postgresql -e POSTGRESQL_DATABASE=db_name -e POSTGRESQL_USER=username -e POSTGRESQL_PASSWORD=password
+----
+
+In case you would like to also set password for the database administrator, append the previous command with:
+----
+-e POSTGRESQL_ADMIN_PASSWORD=admin_pw
+----
+
+To watch the progress of this command type:
+
+----
+$ oc get pods --watch
+----
+
+=== Creating frontend service
+
+To bring your application to OpenShift, you need to specify a repository in which your application lives, using once again the link:../new_app.html[oc new-app] command, in which you will need to specify database related link:../../db_images/postgresql.html#environment-variables[environment variables] we setup in the link:./ruby_on_rails.html#creating-database-service[creating database service] section:
+
+----
+$ oc new-app path/to/source/code --name=rails-app -e POSTGRESQL_USER=username -e POSTGRESQL_PASSWORD=password -e POSTGRESQL_DATABASE=db_name
+----
+
+With this command OpenShift will fetch the source code, setup the Builder image, link:../builds.html[build] your application image and also will deploy the newly created image together with the specified link:../../db_images/postgresql.html#environment-variables[environment variables]. Application is going to be named `rails-app`.
```

Comment:
suggest: "With this command, OpenShift fetches ..., sets up ...., builds your app..., and deploys..."
missing article: "The application is named..."